### ISSUE TITLE
reg_l2_loss not changing at all, despite loss improving?

### ISSUE THREAD

**Issue by klappec-bsci** [google/automl#1046]
If the loss is improving, the weights must be changing? how can the l2 loss not be changing?
any push in the right direction towards debugging this would be greatly appreciated.

I've found that this is the code which generates the L2 loss:
def reg_l2_loss(weight_decay, regex=r'.*(kernel|weight):0$'):
  """Return regularization l2 loss loss."""
  var_match = re.compile(regex)
  return weight_decay * tf.add_n([
      tf.nn.l2_loss(v)
      for v in tf.trainable_variables()
      if var_match.match(v.name)
  ])

100/100 - 210s - det_loss: 10.4960 - cls_loss: 10.1403 - box_loss: 0.0071 - reg_l2_loss: 0.0208 - loss: 10.5169 - learning_rate: 1.0016e-04 - gradient_norm: 10.0000 - val_det_loss: 4.6394 - val_cls_loss: 4.2381 - val_box_loss: 0.0080 - val_reg_l2_loss: 0.0208 - val_loss: 4.6602
Epoch 2/100

100/100 - 161s - det_loss: 0.5200 - cls_loss: 0.1693 - box_loss: 0.0070 - reg_l2_loss: 0.0208 - loss: 0.5409 - learning_rate: 1.0033e-04 - gradient_norm: 5.3011 - val_det_loss: 0.3663 - val_cls_loss: 0.0468 - val_box_loss: 0.0064 - val_reg_l2_loss: 0.0208 - val_loss: 0.3871
Epoch 3/100

100/100 - 161s - det_loss: 0.3011 - cls_loss: 0.0393 - box_loss: 0.0052 - reg_l2_loss: 0.0208 - loss: 0.3219 - learning_rate: 1.0050e-04 - gradient_norm: 2.6325 - val_det_loss: 0.3249 - val_cls_loss: 0.0221 - val_box_loss: 0.0061 - val_reg_l2_loss: 0.0208 - val_loss: 0.3457
Epoch 4/100

100/100 - 162s - det_loss: 0.2823 - cls_loss: 0.0349 - box_loss: 0.0049 - reg_l2_loss: 0.0208 - loss: 0.3031 - learning_rate: 1.0066e-04 - gradient_norm: 2.3600 - val_det_loss: 0.2634 - val_cls_loss: 0.0107 - val_box_loss: 0.0051 - val_reg_l2_loss: 0.0208 - val_loss: 0.2843
Epoch 5/100

100/100 - 161s - det_loss: 0.2527 - cls_loss: 0.0182 - box_loss: 0.0047 - reg_l2_loss: 0.0208 - loss: 0.2735 - learning_rate: 1.0083e-04 - gradient_norm: 2.6735 - val_det_loss: 0.2452 - val_cls_loss: 0.0129 - val_box_loss: 0.0046 - val_reg_l2_loss: 0.0208 - val_loss: 0.2660
Epoch 6/100
